optimum/intel/openvino/configuration.py

```diff
@@ -100,31 +52,122 @@
+        for property_name, new_property_value in zip(self.property_names, self.new_property_values):
+            setattr(self.obj, property_name, new_property_value)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        for property_name, old_property_value in zip(self.property_names, self.old_property_values):
+            setattr(self.obj, property_name, old_property_value)
+
+
+def is_serializable(obj):
+    try:
+        json.dumps(obj)
+        return True
+    except Exception:
+        return False
+
+
+@dataclass
+class OVQuantizationConfigBase(QuantizationConfigMixin):
+    """
+    Base configuration class for quantization parameters
+    """
+
+    def __init__(
+        self,
+        dataset: Optional[Union[str, List[str], nncf.Dataset, datasets.Dataset]] = None,
+        ignored_scope: Optional[Union[dict, nncf.IgnoredScope]] = None,
+        subset_size: Optional[int] = None,
+    ):
+        """
+        Args:
+            dataset (`str or List[str] or nncf.Dataset or datasets.Dataset`, *optional*):
+                 The dataset used for data-aware weight compression or quantization with NNCF.
+            ignored_scope (`dict or nncf.IgnoredScope`, *optional*):
+                An ignored scope that defines the list of model nodes to be ignored during quantization.
+            subset_size (`int`, *optional*):
+                The maximum number of samples composing the calibration dataset.
+        """
+        self.dataset = dataset
+        if isinstance(ignored_scope, dict):
+            ignored_scope = nncf.IgnoredScope(**ignored_scope)
+        self.ignored_scope = ignored_scope
+        self.subset_size = subset_size
+
+    def post_init(self):
+        if not (self.dataset is None or isinstance(self.dataset, (str, list, nncf.Dataset, datasets.Dataset))):
+            raise ValueError(
+                "Dataset must be a instance of either string, list of strings, nncf.Dataset or "
+                f"dataset.Dataset, but found {type(self.dataset)}"
+            )
+        if not (self.ignored_scope is None or isinstance(self.ignored_scope, nncf.IgnoredScope)):
+            raise ValueError(
+                "Ignored scope must be a instance of either dict, or nncf.IgnoredScope but found "
+                f"{type(self.dataset)}"
+            )
+
+    def to_dict_without_properties(self, property_names: Union[List[str], Tuple[str]]) -> Dict[str, Any]:
+        """
+        Calls to_dict() with given properties overwritten with None. Useful for hiding non-serializable properties.
+        """
+        if len(property_names) == 0:
+            return super().to_dict()
+        with replace_properties_values(self, property_names, [None] * len(property_names)):
+            result = super().to_dict()
+        return result
+
+    def to_dict(self) -> Dict[str, Any]:
+        properties_to_omit = [] if is_serializable(self.dataset) else ["dataset"]
+        if isinstance(self.ignored_scope, nncf.IgnoredScope):
```

Comment:
do you think we need it in the configuration (we could create an instance of `IgnoredScope` directly in the quantizer)

Reply:
Agree, done